### ISSUE TITLE
2.2.0.3, binding local volume not working

### ISSUE THREAD

**Issue by achwiej** [docker/for-win#5767]
This is now fixed in 2.2.0.3, so closing this ticket.

_Originally posted by @stephen-turner in https://github.com/docker/for-win/issues/5318#issuecomment-585200795_

Hello, i'm using version 2.2.0.3 but still having the issue with local volumes. 
Here is my diagnostic ID: 83F1FA50-D5B7-4887-825A-1E45B0989D6E/20200213163729

I'm using docker-compose, version:
docker-compose version 1.25.4, build 8d51620a
docker-py version: 4.1.0
CPython version: 3.7.4
OpenSSL version: OpenSSL 1.1.1c  28 May 2019

and my volumes declaration looks like this:
volumes:
    - "C:\\workspace-software\\software-repo\\target\\classes:/software/dev/classes1"
    - "C:\\workspace-software\\software-repo\\target\\classes:/software/dev/classes2"
...

I tried to switch to this version of volume declaration but it does not work any better:
- type: bind
      source: C:\\workspace-software\\software-repo\\target\\classes
      target: /software/dev/classes1
      consistency: consistent
- type: bind
      source: C:\\workspace-software\\software-repo\\target\\classes
      target: /software/dev/classes2
      consistency: consistent

Windows version (using winver):  W10 Pro, 1903 (18362.657)

Please let me know if you need more informations.

**Comment by stephen-turner** [comment]
What is the error or incorrect behaviour that you are seeing?

**Comment by rodrigoscna** [comment]
Also getting the same issue here.

When I enter the container, only the `/application/node_modules` folder exists, and it's populated with the `Dockerfile`'s `npm install` content. Every other file that should be mounted are not there.

The host's `node_modules` folder does not exists or is empty.

Diagnostics ID: 5D73F4B1-9E61-4974-A269-F160538A6946/20200213191235

Versions:
- Windows: 1909 (18363.657)
- Docker: 2.2.0.3 (42716)
- Engine: 19.03.5
- Compose: 1.25.4

Files:

**Dockerfile**:
```Dockerfile
FROM node:12.13-alpine

RUN apk add --no-cache \
  curl \
  git

RUN mkdir -p /application \
  && chown -R node:node /application
WORKDIR /application

USER node

ADD --chown=node:node package.json package-lock.json /application/
RUN npm install \
  && npm cache clean --force

ADD --chown=node:node . /application/

CMD ["npm", "run", "start"]

```

**docker-compose.yaml**
```YAML
version: '3'

services:
  server:
    build:
      context: .
      dockerfile: Dockerfile
    expose:
      - '3000'
    volumes:
      - .:/application
      - /application/node_modules/
    working_dir: /application
    command:
      - npm
      - run
      - start
```

**Comment by djs55** [comment]
Hi @rodrigoscna thanks for your report. I've tried to reproduce what you've described but I'm not sure if I've got the details correct.

Since I'm not familiar with `node` and don't have a working `package.*` etc I tried to simulate this setup with the following:

*docker-compose.yml*
```
version: '3'

services:
  server:
    build:
      context: .
      dockerfile: Dockerfile
    volumes:
      - .:/application
      - /application/node_modules/
    working_dir: /application
```
*Dockerfile*
```
FROM alpine:latest

RUN mkdir /application
ADD . /application

RUN mkdir /application/node_modules
RUN touch /application/node_modules/from-Dockerfile

CMD [ "sleep", "3600000" ]
```
I then start the app:
```
PS C:\Users\djs\bugs\5767> docker-compose up
Starting 5767_server_1 ... done
Attaching to 5767_server_1
```
and then look inside the container:
```
PS C:\Users\djs\bugs\5767> docker exec -it 5767_server_1 sh
/application #
/application # ls
Dockerfile          docker-compose.yml  node_modules
/application # ls node_modules/
from-Dockerfile
/application # touch node_modules/in-VM
/application # ls
Dockerfile          docker-compose.yml  node_modules
/application # ls node_modules/
from-Dockerfile  in-VM
```
And I inspect the container to look at the volume
```
PS C:\Users\djs\bugs\5767> docker inspect 5767_server_1
...
            {
                "Type": "volume",
                "Name": "5275f0d814fd2fc84ea27776eb9cf82f6403417ee52187456188908a1b8294cd",
                "Source": "/var/lib/docker/volumes/5275f0d814fd2fc84ea27776eb9cf82f6403417ee52187456188908a1b8294cd/_data",
                "Destination": "/application/node_modules",
                "Driver": "local",
                "Mode": "",
                "RW": true,
                "Propagation": ""
            }
...
PS C:\Users\djs\bugs\5767> docker run -it --privileged --pid=host justincormack/nsenter1
/ # ls /var/lib/docker/volumes/5275f0d814fd2fc84ea27776eb9cf82f6403417ee52187456188908a1b8294cd/_data
from-Dockerfile  in-VM
```

So what I see is
- the current directory is mounted as `/application`, which is the `WORKDIR`: I can see the `docker-compose.yml`
- the `node_modules` directory is an anonymous volume (stored in `/var/lib/docker/volumes/...`)
- the `node_modules` contains the initial contents from the `Dockerfile` build
- the `node_modules` also contains the extra file I added at runtime `in-VM`

Could you help me understand how your environment differs from mine and/or exactly what you saw versus exactly what you expected? Thanks for all your help!

**Comment by rodrigoscna** [comment]
For some reason, when I tried to replicate your example today, I got a message from Docker that my `D:` drive (where I store my projects) was not shared, even though I set up the sharing since I first configured it.

And now, when I try to share it again, Docker gives me an error message:
```
Cannot change shared directories

Changing shared directories failed, see the log for details.
```

I had to stop all my running containers, and then re-enable the sharing.

Now everything appears to be working.
